### ISSUE TITLE
Resource aws_launchtemplate credit_specification type standard prevents non-t2 instance_types from launching

### ISSUE THREAD

**Issue by Dombo** [terraform-providers/terraform-provider-aws#5081]
### Terraform Version
Terraform v0.11.7
AWS Provider v1.25.0

### Affected Resource(s)

* aws_launch_template
* aws_autoscaling_group

### Terraform Configuration Files

```hcl
resource "aws_launch_template" "unlimited_launchtemplate" {
  name = "${uuid()}"
  image_id = "${var.ami}"
  instance_type = "c5.xlarge"
  key_name = "${var.key_name}"
}

resource "aws_autoscaling_group" "unlimited_asg" {
  name = "unlimited-c5-asg"
  launch_template = {
    name = "${aws_launchtemplate.unlimited_launchtemplate.name}"
  }
  availability_zones = ["${var.availability_zones}"]
  load_balancers = ["${var.loadbalancers}"]
  vpc_zone_identifier = ["${var.subnets}"]
  max_size = "1"
  min_size = "1"
  desired_capacity = "1"
}
```

### Panic Output

The run completes successfully so no panic/debug output.

### Expected Behavior

When specifying a credit_specification of 'standard' - it should be set for the launch_template & attached to the autoscaling_group.

### Actual Behavior

When specifying a credit_specification of 'standard' - instances that are not of class t2.* fail to launch complaining that they do not support credit type unlimited.

### Steps to Reproduce

Using the above config, terraform apply - your run will complete but instances will fail to launch.

1. `terraform apply`

### Important Factoids

I am using c5's which are a fairly new instance class but I don't believe this will change anything.

**Comment by kl4w** [comment]
@Dombo @bflad I don't get an issue when I use the following launch template configuration

```
resource "aws_launch_template" "unlimited_launchtemplate" {
  name = "${uuid()}"
  image_id = "${var.ami}"
  instance_type = "c5.xlarge"
  key_name = "${var.key_name}"
}
```

but I do get an error when I use 
```
resource "aws_launch_template" "unlimited_launchtemplate" {
  name = "${uuid()}"
  image_id = "${var.ami}"
  instance_type = "c5.xlarge"
  key_name = "${var.key_name}"
  
  credit_specification {
    cpu_credits = "standard"
  }
}
```

In the console, launch template creation has the option to not include `credit_specification` in the launch template data.

@bflad how do you think we should best tackle this? via documentation or have a validation function that sets `creditSpecification` to `nil` when the `instance_type` does not match?

**Comment by bflad** [comment]
We just added the following logic to `aws_instance` for this situation, so maybe we should do the same change for `aws_launch_template` as well: https://github.com/terraform-providers/terraform-provider-aws/pull/5114/files

**Comment by bflad** [comment]
This has been released in [version 1.28.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1280-july-18-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
